.circleci/config.yml

```
@@ -182,6 +186,24 @@ jobs:
                 name: coveralls
                 command: npm run coveralls
 
+  check_coding_style:
+    docker:
+      - image: cimg/node:current
+    steps:
+      - show-npm-version
+      - checkout
+      - shellcheck/install
+      - install-dependencies:
+          cache-id: solc-js
+      - run:
+          name: Check for javascript/typescript coding style
+          command: npm run lint
+      - shellcheck/check:
+          ignore-dirs: |
+            ./.git
+            ./node_modules
+            ./dist
```

Isn't `./` superfluous? On the other hand I'd add `/` at the end to make it clear we expect these to be dirs.
```suggestion
            .git/
            node_modules/
            dist/
```

---

Yes. I was wondering if we indeed need to exclude those directories because they will probably not contain any shell script. But to avoid the orb to even parsing those directories, I preferred to explicitly them.

---

You could also create a separate job for shellcheck. Then you would not have to install the JS dependencies in it. Also, the shellcheck job would probably complete faster.

By the way, you probably don't need to ignore `dist/` because you're not running `build`.